internal/provider/provider.go

```diff
@@ -240,3 +268,40 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 		return md, nil
 	}
 }
+
+// getDefaultAuthMethodId iterates over boundary client.List() to find the default auth method ID for the given scopeId.
+// If there is only one auth method, it'll return it even if it's not the primary auth method
+// If scope ID is empty or no primary auth method is found, it returns an error.
+func getDefaultAuthMethodId(ctx context.Context, client *authmethods.Client, scopeId, amType string) (string, error) {
```

Comment:
Do we want to support ldap auth methods in this func?

Reply:
Yeah, I think we can add support for that as well. A separate PR where we add LDAP support and also use the default primary method for that.